docs/writing-tests/testdriver.md

```diff
@@ -222,3 +222,60 @@ Note that if an action uses an element reference, the context will be
 derived from that element, and must match any explicitly set
 context. Using elements in multiple contexts in a single action chain
 is not supported.
+
+## BiDi
+
+The api in `test_driver.bidi` provides access to the
+[WebDriver BiDi](https://w3c.github.io/webdriver-bidi\>) protocol.
+
+### Context
+
+A WebDriver BiDi "browsing context" is equivalent to an
+[HTML navigable](https://html.spec.whatwg.org/multipage/document-sequences.html#navigable).
+In WebDriver BiDi, you can interact with any browsing context, regardless of whether
+it's currently active. You can target a specific browsing context using either its
+unique string ID or its `WindowProxy` object.
+
+```eval_rst
+:Context: (*String|WindowProxy*)  A browsing context. Can be specified by its ID 
+          (a string) or using a `WindowProxy` object.
+```
+
+### Events
+
+To receive WebDriver BiDi [events](https://w3c.github.io/webdriver-bidi/#events), you
+need to subscribe to them. Events are only emitted for browsing contexts with an
+active subscription. You can also create a global subscription to receive events from
+all the contexts.
+
+#### Events Buffer ####
+
+If there are
+[buffered events](https://w3c.github.io/webdriver-bidi/#log-event-buffer), they will
+be emitted before the `subcsribe` command's promise is resolved.
+
+### Log Module
+
+This module corresponds to the WebDriver BiDi
+[Log](https://w3c.github.io/webdriver-bidi/#module-log) module and provides access to
+browser logs.
+
+#### Entry added
+
+This provides methods to subscribe to and listen for the
+[`log.entryAdded`](https://w3c.github.io/webdriver-bidi/#event-log-entryAdded) event.
+
+**Example:**
+
+```javascript
+await test_driver.bidi.log.entry_added.subscribe();
+const log_entry_promise = test_driver.bidi.log.entry_added.once();
+console.log("some message");
+const event = await log_entry_promise;
+```
+
+```eval_rst
```

Comment:
I think we should update the API section with new APIs and update the "Using test_driver in other browsing contexts" with WebDriver BiDi specific functions instead of writing a completely new section. Also, let's use the full name "WebDriver BiDi" when referring to the WebDriver BiDi specification.

Reply:
> I think we should update the API section with new APIs and update the "Using test_driver in other browsing contexts" with WebDriver BiDi specific functions instead of writing a completely new section. 

"Using test_driver in other browsing contexts" is refers to the "action API", and it's quite different from BiDi.

> Also, let's use the full name "WebDriver BiDi" when referring to the WebDriver BiDi specification.

I guess you refer to the section name `## BiDi`, which was used because of the property explosed in the API: `test_driver.bidi`. I still can change it if you insist.